### PR TITLE
Update doc Makefile

### PR DESCRIPTION
**Pull request by liuzhe-lz**
Commands:

  - `make`: Build English doc to `build/html` and Chinese doc to `build/html_zh`.
  - `make en`: Build English doc only.
  - `make zh`: Build Chinese doc only.
  - `make clean`: Remove built docs.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
+	## English ##
+	sphinx-build -T source build
 
-# Put it first so that "make" without argument is like "make help".
-help:
-	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+.PHONY: zh
+zh:
+	## Chinese ##
+	sphinx-build -T -D language=zh source build_zh
```
Comment (matluster) [inline docs/Makefile:12]: Add `build_zh` to `gitignore`.

File: docs/Makefile [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-SOURCEDIR     = source
-BUILDDIR      = build
+.PHONY: en
+en:
+	## English ##
+	sphinx-build -T source build
 
-# Put it first so that "make" without argument is like "make help".
-help:
-	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (matluster) [inline docs/Makefile:12]: Is pipeline happy about this?

**Comment by matluster** [comment]
There's a linkcheck in v2.7 branch. Let's wait for release.

File: docs/Makefile [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-.PHONY: help Makefile
-
-# Catch-all target: route all unknown targets to Sphinx using the new
-# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile
-	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+.PHONY: clean
+clean:
+	rm -rf build
+	rm -rf source/reference/_modules
```
Comment (J-shang) [inline docs/Makefile:17]: we still have `_modules` now?

Comment (liuzhe-lz) [reply docs/Makefile:17]: I'm adding it back.

**Review by J-shang — APPROVED** [review]

**Review by ultmaster — APPROVED** [review]

**Pull request merged**